File: app/src/main/java/kapil/circularlayoutmanager/RecyclerItemDecoration.java
package kapil.circularlayoutmanager;

import android.content.Context;
import android.graphics.Rect;
import android.support.v7.widget.RecyclerView;
import android.util.TypedValue;
import android.view.View;

/**
 * Item decorator for recycler view. Adds margin to the outermost children of the list to induce an
 * over scroll effect with drag.
 */

class RecyclerItemDecoration extends RecyclerView.ItemDecoration {
    @Override
    public void getItemOffsets(Rect outRect, View view, RecyclerView parent, RecyclerView.State state) {
        outRect.top = (int) DpToPx(parent.getContext(), 5);
        outRect.bottom = (int) DpToPx(parent.getContext(), 5);

        if (parent.getChildAdapterPosition(view) == 0) {
            outRect.top = (int) DpToPx(parent.getContext(), 100);
        }
        if (parent.getChildAdapterPosition(view) == parent.getAdapter().getItemCount() - 1) {
            outRect.bottom = (int) DpToPx(parent.getContext(), 100);
        }
    }

    private float DpToPx(Context context, float dp) {
        return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, context.getResources().getDisplayMetrics());
    }
}

File: app/src/main/java/kapil/circularlayoutmanager/RecyclerItemDecoration.kt
package kapil.circularlayoutmanager

import android.graphics.Rect
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ItemDecoration

/**
 * Item decorator for recycler view. Adds margin to the outermost children of the list to induce an
 * over scroll effect with drag.
 *
 * Setting the top and bottom item offsets for the boundary items (header and footer margins), as
 * done below, would be helpful in tuning desired effect specially when Auto Stabilization feature
 * is enabled in [com.kapil.circularlayoutmanager.CircularLayoutManagerNew].
 */
class RecyclerItemDecoration : ItemDecoration() {

    override fun getItemOffsets(
        outRect: Rect,
        view: View,
        parent: RecyclerView,
        state: RecyclerView.State
    ) {
        outRect.top = parent.context.resources.getDimension(R.dimen.margin_xs).toInt()
        outRect.bottom = parent.context.resources.getDimension(R.dimen.margin_xs).toInt()
        if (parent.getChildAdapterPosition(view) == 0) outRect.top =
            parent.context.resources.getDimension(R.dimen.margin_xxxl).toInt()
        if (parent.getChildAdapterPosition(view) == parent.adapter!!.itemCount - 1)
            outRect.bottom = parent.context.resources.getDimension(R.dimen.margin_xxxl).toInt()
    }
}